docs/source/deployment/lambda.md

```diff
@@ -58,6 +59,226 @@ exports.graphqlHandler = server.createHandler();
+Transform: AWS::Serverless-2016-10-31
+
+Resources:
+  GraphQLFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      CodeUri: graphql/
+      Handler: app.graphqlHandler
+      Timeout: 5 #
+      MemorySize: 256
+      Runtime: nodejs14.x
+      Events:
+        AutomaticPersistedQueries:
+          Type: Api
+          Properties:
+            Path: /
+            Method: get
+        GraphQL:
+          Type: Api
+          Properties:
+            Path: /
+            Method: post
+        ApolloSandbox:
+          Type: Api
+          Properties:
+            Path: /
+            Method: options
+```
+
+You will need to ensure `options` are exposed to use Apollo Explorer and Apollo Sandbox.
+
+*Note: The example above is using v1 of AWS API Gateway which is denoted by the `Type: Api`, but v2 (`Type: HttpApi`) is supported as well. The `apollo-server-lambda` [repository currently has tests that cover both v1 and v2 API Gateway events and ALB events](https://github.com/apollographql/apollo-server/blob/main/packages/apollo-server-lambda/src/__tests__/lambdaApollo.test.ts). More info about API Event Source can be found [here](https://github.com/aws/serverless-application-model/blob/master/versions/2016-10-31.md#api)*
+
+### Invoking SAM locally
+
+To run your project locally with SAM, you will need to provide an event payload that "invokes" the AWS Lambda handler running locally in docker on your machine. [Invoking your lambda function](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-using-invoke.html) locally can be done with the SAM CLI:
+
+```
+sam local invoke "GraphQLFunction" -e event.json
+```
+
+For the `event.json`, you will need to provide the proper event depending on what is being used to provide the event to the AWS lambda. This will most commonly used will be an [AWS API Gateway, either v1 or v2](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html).
+
+Below is an example of the minimum `event.json` needed for a standard GraphQL request when using AWS API Gateway v1:
+
+```
+{
+  "requestContext": {
+    "version": "1.0"
+  },
+  "path": "/graphql",
+  "httpMethod": "POST",
+  "body": "{ \"query\": \"{__typename}\" }",
+  "multiValueHeaders": {
+    "Accept": "*",
+    "origin": "",
+    "content-type": "application/json"
+  }
+}
+```
+
+Below is an example of the minimum `event.json` needed for a standard GraphQL request when using AWS API Gateway v2:
+
+```
+{
+  "version": "2.0",
+  "rawPath": "/graphql",
+  "requestContext": {
+    "http": {
+      "method: "POST",
+      "path": "/graphql",
+    }
+  },
+  "body": "{ \"query\": \"{__typename}\" }",
+  "headers": {
+    "Accept": "*",
+    "origin": "",
+    "content-type": "application/json"
+  }
+}
+```
+
+Below is an example of the minimum `event.json` needed to test an Automatic Persisted Queries (APQ) GraphQL request when using v1 API Gateway:
```

Comment:
I think this doc may make people confused into thinking that the APQ feature is more tied into the choice of HTTP method than it actually is.

I'm also curious why we think documenting how to do APQs manually is helpful.  Doing APQs manually is pretty tough since you need to calculate the hash!

If you do leave all these examples in, I suggest hiding some of them in `<ExpansionPanel>`s (which requires you to change the filename to lambda.mdx). Also link to the APQ page (`../performance/apq/`)

Reply:
Honestly I only documented this part because of the [curl testing docs](https://www.apollographql.com/docs/apollo-server/performance/apq/#command-line-testing) and you can't do that unless you deploy the lambda. I'm fine with just removing these.